mono/metadata/object.c

```diff
@@ -6486,16 +6486,7 @@ mono_string_new_wrapper (const char *text)
 {
 	MONO_REQ_GC_UNSAFE_MODE;
 
-	MonoDomain *domain = mono_domain_get ();
-
-	if (text) {
-		ERROR_DECL (error);
-		MonoString *result = mono_string_new_checked (domain, text, error);
-		mono_error_assert_ok (error);
-		return result;
-	}
-
-	return NULL;
+	return mono_string_new (mono_domain_get (), text);
```

Comment:
This should result in a compilation error about calling a `MONO_RT_EXTERNAL_ONLY` function.

Reply:
Argh. I work on Windows where that doesn't error.